awx/sso/backends.py

```diff
@@ -395,10 +395,17 @@ def on_populate_user(sender, **kwargs):
         remove = bool(team_opts.get('remove', True))
         _update_m2m_from_groups(user, ldap_user, team.member_role.members, users_opts, remove)
 
+    # Check if user.profile is available, otherwise force user.save()
+    try:
+        _ = user.profile
+    except ValueError:
+        force_user_update = True
+    finally:
+        if force_user_update:
+            user.save()
+
     # Update user profile to store LDAP DN.
-    if force_user_update:
-        user.save()
-        profile = user.profile
-        if profile.ldap_dn != ldap_user.dn:
-            profile.ldap_dn = ldap_user.dn
-            profile.save()
+    profile = user.profile
```

Comment:
Can this `user.profile` result in a `ValueError`? Or will the `save()` above ensure that `user.profile` gets set?

Reply:
@chrismeyersfsu  during my tests, the only time I was able to get a `ValueError` was for a brand new user. For users already present in the database,  `user.profile`  worked as expected.